src/sinks/splunk_hec/metrics.rs

```diff
@@ -0,0 +1,713 @@
+use crate::{
+    config::{DataType, GenerateConfig, SinkConfig, SinkContext, SinkDescription},
+    event::{Event, Metric, MetricValue},
+    internal_events::SplunkInvalidMetricReceived,
+    internal_events::{SplunkEventEncodeError, SplunkEventSent},
+    sinks::splunk_hec::conn,
+    sinks::util::{encode_namespace, http::HttpSink, BatchConfig, Compression, TowerRequestConfig},
+    sinks::{Healthcheck, VectorSink},
+    template::Template,
+    tls::TlsOptions,
+};
+use http::Request;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::iter;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct HecSinkMetricsConfig {
+    pub default_namespace: Option<String>,
+    pub token: String,
+    pub endpoint: String,
+    #[serde(default = "host_key")]
+    pub host_key: String,
+    pub index: Option<Template>,
+    pub sourcetype: Option<Template>,
+    pub source: Option<Template>,
+    #[serde(default)]
+    pub compression: Compression,
+    #[serde(default)]
+    pub batch: BatchConfig,
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+    pub tls: Option<TlsOptions>,
+}
+
+#[derive(Serialize, Debug, PartialEq)]
+#[serde(untagged)]
+enum FieldValue<'a> {
+    Str(Cow<'a, str>),
+    Float(f64),
+}
+
+impl<'a> From<&'a str> for FieldValue<'a> {
+    fn from(s: &'a str) -> Self {
+        FieldValue::Str(Cow::Borrowed(s))
+    }
+}
+
+impl<'a> From<String> for FieldValue<'a> {
+    fn from(s: String) -> Self {
+        FieldValue::Str(Cow::Owned(s))
+    }
+}
+
+impl<'a> From<f64> for FieldValue<'a> {
+    fn from(f: f64) -> Self {
+        FieldValue::Float(f)
+    }
+}
+
+type FieldMap<'a> = BTreeMap<&'a str, FieldValue<'a>>;
+
+#[derive(Serialize, Debug)]
+struct HecEvent<'a> {
+    time: f64,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    host: Option<String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    index: Option<String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    source: Option<String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sourcetype: Option<String>,
+
+    fields: FieldMap<'a>,
+
+    event: &'a str,
+}
+
+fn host_key() -> String {
+    crate::config::log_schema().host_key().to_string()
+}
+
+inventory::submit! {
+    SinkDescription::new::<HecSinkMetricsConfig>("splunk_hec_metrics")
+}
+
+impl GenerateConfig for HecSinkMetricsConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(Self {
+            default_namespace: None,
+            token: "${VECTOR_SPLUNK_HEC_TOKEN}".to_owned(),
+            endpoint: "endpoint".to_owned(),
```

Comment:
```suggestion
            endpoint: "http://localhost:8088".to_owned(),
```

I think it's better to use a syntactically valid default.